arbnode/node.go

```diff
@@ -564,7 +564,17 @@ func StakerDataposter(
 	dpCfg := func() *dataposter.DataPosterConfig {
 		return &cfg.Staker.DataPoster
 	}
-	return dataposter.NewDataPoster(db, l1Reader, transactOpts, redisC, redisLock, dpCfg, mdRetriever)
+	return dataposter.NewDataPoster(
+		&dataposter.DataPosterOpts{
+			Database:          db,
+			HeaderReader:      l1Reader,
+			Auth:              transactOpts,
+			RedisClient:       redisC,
+			RedisLock:         redisLock,
+			Config:            dpCfg,
+			MetadataRetriever: mdRetriever,
+			RedisKey:          "staker-data-poster.queue",
```

Comment:
We should put the address as part of that, because we run multiple stakers and it's easier to use the same redis for both.

Reply:
I'm on my phone but I think the validator L1 auth is a variable where you can get the From address. It might be nil, in which case we can use the zero address, or with your new no-op validator wallet PR, we shouldn't need to create a data poster at all in that case.

Reply:
Done. PTAL.